lib/graphql/client.ts

```diff
@@ -42,15 +42,17 @@ export class GraphQLClient {
       [key: string]: any;
     },
   ): Promise<RawQueryResponse> => {
-    if (!this.auth.tokenManager.token) {
-      throw new UserUnauthorizedError();
+    if (this.auth) {
```

Comment:
hi @zlu-kontist , this PR is for making `auth` optional, hence to make request to public endpoint.

so if `!this.auth`, it will not include authorization header in the request.

Reply:
I guess I'm a bit confused.  if this.auth doesn't exsit, why should we include an auth header?

Reply:
I mean if `this.auth` does not exist, we wouldn't include an auth header.

Reply:
Actually, would it make sense to leverage factory pattern here by scoping requests without auth to only public apis and require auth for all private apis.

Reply:
actually, I just implement from what @johannespfeiffer suggested 😅

@johannespfeiffer wdyt?

Reply:
Scoping requests without auth is not easy, as the schema is autogenerated from GraphQL.

Reply:
If we want to reduce code duplication and number of `if`s we could use a helper method like this. (not sure if it is still easy enough to read)

```
  private getAuthHeader = (): [string, string] | false => {
    if (!this.auth) {
      return false;
    }

    if (!this.auth.tokenManager.token) {
      throw new UserUnauthorizedError();
    }

    return ["Authorization", `Bearer ${this.auth.tokenManager.token.accessToken}`];
  }
```

and then use it with
```
    const auth = this.getAuthHeader();
    auth && this.client.setHeader(...auth);
```

respective
```
    const auth = this.getAuthHeader();
    const connectionParams = auth ? { [auth[0]]: auth[1] } : {};
```

Reply:
very nice, thanks!